task_processing/plugins/mesos/mesos_executor.py

```diff
@@ -40,7 +40,8 @@ def valid_volumes(volumes):
 class MesosTaskConfig(PRecord):
     uuid = field(type=uuid.UUID, initial=uuid.uuid4)
     name = field(type=str, initial="default")
-    image = field(type=str, initial="ubuntu:xenial")
+    # image is optional for the mesos containerizer
```

Comment:
can we add an invariant here to enforce this?

Reading the docs, we can have something like:

```
class MesosTaskConfig(PRecord):
    __invariant__ = lambda config: (config.image is not None if config.containerizer is not 'MESOS', 'Image required for chosen containerizer')
    ....
```